sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala

```diff
@@ -201,6 +202,12 @@ private[hive] class SparkExecuteStatementOperation(
       parentSession.getUsername)
     setHasResultSet(true) // avoid no resultset for async run
 
+    if (queryTimeout > 0) {
+      Executors.newSingleThreadScheduledExecutor.schedule(new Runnable {
+          override def run(): Unit = cancel(OperationState.TIMEDOUT)
```

Comment:
Adding a log?

Reply:
okay

Reply:
Did you mean the log (that I added in L206) before starting the thread? Looks `cancel` itself already has a log in https://github.com/apache/spark/pull/29933/files#diff-72dcd8f81a51c8a815159fdf0332acdcR343 though.

Reply:
We only know from the log in `cancel` that the query is canceled, but don't know it is timeout.

Reply:
I'd like to let user know it is canceled because of timeout.

Reply:
Updated